### ISSUE TITLE
dash/sh: warn about using parameter-array, e.g. "${@:2}"

### ISSUE THREAD

**Issue by bittorf** [koalaman/shellcheck#815]
#### For bugs
- Rule Id: ???
- [x ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/dash or /bin/sh
set -- A B C D E F
echo "${@:2}"

```

#### Here's what shellcheck currently says:
<no problem>

#### Here's what I wanted or expected to see:
can not be used with dash/sh
